.github/workflows/smoketest.yml

```diff
@@ -0,0 +1,432 @@
+          libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev nodejs libevent-dev libatlas-base-dev libblas-dev python3-dev \
+          default-libmysqlclient-dev libsqlite3-dev libmemcached-dev libcurl4-openssl-dev pkg-config libtool autotools-dev automake \
+          libssl-dev
+
+    - name: run
+      run: |
+        export LANG="en_US.UTF-8"
+        sudo update-locale
+        tar --zstd -xf install.tgz
+        source install/opt/HPCCSystems/sbin/hpcc_setenv
+        install/opt/HPCCSystems/etc/init.d/hpcc-init start
+        cd install/opt/HPCCSystems/testing/regress
+
+        (cd ecl; ls ${{ matrix.match.pattern }}) > matches.tmp
+        echo queries are:
+        cat matches.tmp
+        if [[ -z "${{ matrix.match.exclude }}" ]]; then \
+          queries="$(cat matches.tmp | tr '\n' ' ')"; \
+        else \
+          queries="$(cd ecl; ls ${{ matrix.match.exclude }} | grep -v -f - ../matches.tmp | tr '\n' ' ')"; \
+        fi
+        echo queries after exclude: ${queries}
+
+        ./ecl-test query --pq 2 --target ${{ matrix.engine }} --excludeclass python2,embedded-r,embedded-js,3rdpartyservice,spray ${queries}
+        grep Failure: ${{ github.workspace }}/regress/log/${{ matrix.engine }}.*.log
+        if [[ "$(grep -oP '(?<=^Failure: )[0-9]+$' ${{ github.workspace }}/regress/log/${{ matrix.engine }}.*.log)" -gt 0 ]]; then exit 1; fi
+
+    - name: logs-on-failure
+      if: ${{ failure() }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: regress-fail-logs-${{ matrix.engine }}-${{ steps.vars.outputs.matrix-setname }}
+        path: |
+          install/var/log/HPCCSystems
+          regress/
+        if-no-files-found: error
+
+
+  # NB: this doesn't really need the post-setup data files included in install.tgz
+  # but as this is relatively quick and in parallel with others, it probably doesn't matter
+  unittests:
+    needs: build-and-setup
+    if: needs.build-and-setup.outputs.cache-hit != 'true' && github.event_name == 'pull_request'
+    timeout-minutes: 10
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: installsetup-artifact
+
+    - name: Prerequisites
+      run: |
+        sudo apt-get update
+        sudo apt-get -yq install bison flex build-essential binutils-dev libldap2-dev libcppunit-dev libicu-dev libxslt1-dev \
+          zlib1g-dev libboost-regex-dev libarchive-dev libv8-dev default-jdk libapr1-dev libaprutil1-dev libiberty-dev \
+          libhiredis-dev libtbb-dev libxalan-c-dev libnuma-dev nodejs libevent-dev libatlas-base-dev libblas-dev python3-dev \
+          default-libmysqlclient-dev libsqlite3-dev libmemcached-dev libcurl4-openssl-dev pkg-config libtool autotools-dev automake \
+          libssl-dev
+
+    - name: run
+      run: |
+        export LANG="en_US.UTF-8"
+        sudo update-locale
+        tar --zstd -xf install.tgz
+        install/opt/HPCCSystems/bin/unittests
```

Comment:
Perhaps it would be nice to capture unittests output and add it to artifact if it is failed.  (In same way as it happens with regression tests)

Reply:
@AttilaVamos - The stdout/stderr output is captured from all steps, and viewable under the action, by expanding the step, or you can search the logs from the action or 'download log archive' or 'view raw logs'.

The 'logs-on-failure' step of the regression suite job(s) capture generated log files, but unittests don't generate any log files, only stdout/stderr logging that is captured already.